### ISSUE TITLE
Failed to fix 'passing a unit value to a function'

### ISSUE THREAD

**Issue by nicktrandafil** [rust-lang/rust-clippy#14857]
### Summary

`cargo clippy --fix` output:
```
warning: passing a unit value to a function
   --> src/lib.rs:209:23
    |
209 |         new_input_tx: watch::channel(Default::default()).0,
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unit_arg
    = note: `#[warn(clippy::unit_arg)]` on by default
help: move the expression in front of the call and replace it with the unit literal `()`
    |
209 ~         new_input_tx: {
210 +             Default::default();
211 +             watch::channel(())
212 ~         }.0,
    |

after fixes were automatically applied the compiler reported errors within these files:

  * src/lib.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust-clippy/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0790]: cannot call associated function on trait without specifying the corresponding `impl` type
   --> src/lib.rs:210:13
    |
210 |             Default::default();
    |             ^^^^^^^^^^^^^^^^^^ cannot call associated function of trait
    |
help: use a fully-qualified path to a specific available implementation
    |
210 |             </* self type */ as Default>::default();
    |             +++++++++++++++++++        +

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0790`.
Original diagnostics will follow.
```

### Reproducer

I tried this code:

```rust
<code>
```

I expected to see this happen:

Instead, this happened:

### Version

```text

```

**Comment by profetia** [comment]
I think we should just remove the expr if it is `Default:: default`
